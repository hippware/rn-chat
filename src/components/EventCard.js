import React from "react";
import {Text, View, StyleSheet, Image, TouchableOpacity} from "react-native";
import Card from './Card';
import CardText from './CardText';
import Avatar from './Avatar';
import {k} from './Global';
import ResizedImage from './ResizedImage';
import { Actions } from 'react-native-router-native';
import Profile from '../model/Profile';
import Chats from '../model/Chats';
import EventChat from '../model/EventChat';
import Chat from '../model/Chat';
import {observer} from "mobx-react/native";
import location from '../store/location';
import statem from '../../gen/state';

@observer
export default class EventCard extends React.Component {
  render(){
    const isDay = location.isDay;
    const event = this.props.item;
    return <View/>
    return (
      <Card style={[{marginTop:10}, this.props.style]}
            isDay={isDay}
            onPress={this.props.onPress}
            innerStyle={{paddingTop:20*k,paddingLeft:1,paddingRight:1,paddingBottom:10*k}}
            footer={
                        <View style={{position:'absolute',top:0,left:30*k,right:0,height:40*k}}>
                          <View style={{flex:1, flexDirection:'row'}}>
                          </View>

                            {this.props.onPostOptions && <TouchableOpacity ref='button' onPress={e=>this.props.onPostOptions(e, this.refs.button)}
                                style={{position:'absolute', flexDirection:'row',  backgroundColor:'transparent', alignItems:'center', top:20*k, right:20*k}}>
                                <Text style={{fontFamily:'Roboto-Light',fontSize:12, color:'rgb(155,155,155)'}}>{event.date} </Text>
                                <Image source={require("../../images/iconPostOptions.png")}/>
                            </TouchableOpacity>}
                            {!this.props.onPostOptions && <View style={{position:'absolute', backgroundColor:'transparent', flexDirection:'row', alignItems:'center', top:20*k, right:5*k}}>
                                    <Text style={{fontFamily:'Roboto-Light',fontSize:12*k, color:'rgb(155,155,155)'}}>{event.date}</Text>
                                </View>
                                }
                        </View>
                        }>
        {this.props.children}
      </Card>
    );
  }
}
