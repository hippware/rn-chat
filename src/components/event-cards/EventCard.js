// @flow

import React from 'react';
import Card from '../Card';
import {k} from '../Global';
import {observer} from 'mobx-react/native';
import EventBotCard from './EventBotCard';
import EventBotShareCard from './EventBotShareCard';
import EventBotNoteCard from './EventBotNoteCard';
import EventBotPostCard from './EventBotPostCard';
import {getType, isAlive} from 'mobx-state-tree';

type Props = {
  item: any,
};

const eventCardMap = {
  EventBotCreate: EventBotCard,
  EventBotPost: EventBotPostCard,
  EventBotShare: EventBotShareCard,
  EventBotNote: EventBotNoteCard,
};
@observer
export default class EventCard extends React.Component {
  props: Props;
  card: any;

  render() {
    const row = this.props.item;
    if (!isAlive(row)) {
      return null;
    }
    const CardClass = eventCardMap[getType(row).name];
    let profile;
    try {
      // TODO: deleted bot throws an error here trying to generate a profile from a bad id
      profile = row.target;
      if (!profile || !profile.id) {
        return null;
      }
    } catch (err) {
      console.log('TODO: fix bot delete', err);
      return null;
    }

    return (
      <Card
        key={row.id}
        onPress={() => this.card.onPress && this.card.onPress()}
        style={{
          paddingTop: 10 * k,
          paddingLeft: 0,
          paddingRight: 0,
          paddingBottom: 0,
        }}
      >
        <CardClass ref={r => (this.card = r)} item={row} />
      </Card>
    );
  }
}
