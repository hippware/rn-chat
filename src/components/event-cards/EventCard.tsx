import React from 'react'
// import EventBotShareCard from './EventBotShareCard'
import {getType} from 'mobx-state-tree'
import {IEvent, IEventBotInvite, IEventBotPost, IEventBotGeofence} from 'wocky-client'
import EventFriendInviteRequestCard from './EventFriendInviteRequestCard'
import {observer} from 'mobx-react'
import {Actions} from 'react-native-router-flux'
import EventCardTemplate from './EventCardTemplate'
import {IEventLocationShare} from 'third-party/wocky-client/src/model/EventLocationShare'
import {RText, GradientButton} from '../common'
import {TouchableOpacity} from 'react-native'
import {colors} from 'src/constants'

const EventBotInviteCard = observer(
  ({
    item: {bot, relativeDateAsString, sender, isAccepted, isResponse},
  }: {
    item: IEventBotInvite
  }) => (
    <EventCardTemplate
      profile={sender}
      iconType="geo"
      timestamp={relativeDateAsString}
      action={
        isResponse
          ? isAccepted
            ? 'accepted your invite to'
            : 'rejected your invite to'
          : 'invited you to follow'
      }
      line2={bot.title}
      onPress={() => Actions.botDetails({botId: bot.id})}
    />
  )
)

export const EventBotPostCard = observer(
  ({item: {bot, relativeDateAsString, post}}: {item: IEventBotPost}) => {
    if (!post.profile) {
      return null
    }
    return (
      <EventCardTemplate
        profile={post.profile!}
        iconType="notification"
        timestamp={relativeDateAsString}
        action="commented on"
        line2={bot.title}
        onPress={() => Actions.botDetails({botId: bot.id})}
      />
    )
  }
)

const EventBotGeofenceCard = observer(
  ({item: {bot, relativeDateAsString, profile, isEnter}}: {item: IEventBotGeofence}) => (
    <EventCardTemplate
      profile={profile}
      iconType="geo"
      timestamp={relativeDateAsString}
      action={isEnter ? 'is at' : 'left'}
      line2={bot.title}
      onPress={() => Actions.botDetails({botId: bot.id})}
    />
  )
)

const Button = ({
  text,
  onPress,
  textStyle,
  style,
}: {
  text: string
  onPress: any
  textStyle?: any
  style?: any
}) => (
  <TouchableOpacity
    style={[
      {
        borderColor: colors.PINK,
        borderWidth: 1,
        marginTop: 5,
        height: 29,
        borderRadius: 3,
        justifyContent: 'center',
      },
      style,
    ]}
    onPress={onPress}
  >
    <RText
      weight="Medium"
      color={colors.PINK}
      style={[{marginLeft: 10, marginRight: 10}, textStyle]}
    >
      {text}
    </RText>
  </TouchableOpacity>
)

const EventLocationShareCard = observer(
  ({item: {sharedWith, relativeDateAsString}}: {item: IEventLocationShare}) => (
    <EventCardTemplate
      profile={sharedWith}
      iconType="share"
      timestamp={relativeDateAsString}
      action={'is sharing location with you'}
    >
      {sharedWith.sharesLocation ? (
        sharedWith.receivesLocationShare ? (
          <GradientButton
            text="SHARING LOCATION"
            style={{width: 160, height: 29, borderRadius: 4, marginVertical: 4}}
            textStyle={{fontSize: 12, color: 'white'}}
            onPress={() => Actions.liveLocationCompose({profile: sharedWith})}
          />
        ) : (
          <Button
            text="SHARE YOUR LOCATION"
            onPress={() => Actions.liveLocationCompose({profile: sharedWith})}
          />
        )
      ) : (
        undefined
      )}
    </EventCardTemplate>
  )
)

const eventCardMap: {[key: string]: any} = {
  EventBotPost: EventBotPostCard,
  // EventBotShare: EventBotShareCard,
  EventBotGeofence: EventBotGeofenceCard,
  EventFriendInvite: EventFriendInviteRequestCard,
  EventBotInvite: EventBotInviteCard,
  EventLocationShare: EventLocationShareCard,
}

type Props = {
  item: IEvent
}
const EventCard = observer(({item}: Props) => {
  const CardClass = eventCardMap[getType(item).name]
  return CardClass ? <CardClass item={item} /> : null
})

export default EventCard
