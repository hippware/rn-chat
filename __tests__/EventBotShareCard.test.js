// @flow

import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import EventCard from '../src/components/event-cards/EventBotCard'
import {Provider} from 'mobx-react/native'
import {EventBotShare, Bot, Message} from 'wocky-client'
import wocky from './utils/mockWocky'
import mockLocationStore from './utils/mockLocationStore'

describe('EventBotShareCard', () => {
  it('renders', async () => {
    await wocky.login('user', 'password', 'host')
    const owner = wocky.profiles.get('user', {handle: 'test', avatar: null})
    const bot = Bot.create({
      id: '1234',
      isSubscribed: false,
      title: 'title',
      server: 'server',
      radius: 30,
      owner,
      // image: FileRef,
      description: 'description',
      public: true,
      location: {latitude: 1, longitude: 1},
      address: 'address',
      followersSize: 0,
      totalItems: 0,
      // addressData: types.optional(Address, {}),
      // subscribers: types.optional(ProfilePaginableList, {}),
      // posts: types.optional(BotPostPaginableList, {}),
      error: '',
    })

    mockLocationStore.setPosition({longitude: 2, latitude: 2, accuracy: 2})

    // TODO: create different profile as owner
    const message = Message.create({id: '789', time: 789, archiveId: '789', from: owner.id})
    const shareBot = EventBotShare.create({
      id: '456',
      cursor: '456',
      time: 456,
      bot,
      message,
      action: 'share',
    })
    const toRender = (
      <Provider locationStore={mockLocationStore}>
        <EventCard item={shareBot} />
      </Provider>
    )
    const tree = renderer.create(toRender).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
