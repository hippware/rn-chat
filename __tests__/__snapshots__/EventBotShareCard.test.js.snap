// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventBotShareCard renders 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "borderBottomWidth": 1,
          "borderColor": "rgb(212,212,212)",
          "flexDirection": "row",
          "paddingHorizontal": 30,
          "paddingVertical": 20,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "justifyContent": "flex-end",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "marginRight": 30,
            },
            Object {
              "height": 144,
              "width": 144,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 72,
                "borderWidth": 4,
                "height": 144,
                "width": 144,
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "rgb(228,228,228)",
                "borderColor": "rgb(255,255,255)",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "rgb(63,50,77)",
                  "fontFamily": "Roboto-Regular",
                },
                Object {
                  "fontSize": 36,
                },
              ]
            }
          >
            T
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "borderColor": "white",
                "position": "absolute",
              },
              Object {
                "borderRadius": 14.4,
                "borderWidth": 2.88,
                "height": 28.8,
                "left": 108,
                "top": 108,
                "width": 28.8,
              },
              Object {
                "backgroundColor": "rgb(211,211,211)",
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "rgb(63,50,77)",
                      "fontFamily": "Roboto-Medium",
                      "fontSize": 13,
                    },
                    Object {
                      "letterSpacing": -0.1,
                    },
                  ]
                }
              >
                @
                test
              </Text>
            </View>
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "rgb(114,100,109)",
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 13,
                },
                Object {
                  "letterSpacing": -0.1,
                },
              ]
            }
          >
             
            created
          </Text>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "rgb(155,155,155)",
                "fontFamily": "Roboto-Light",
                "fontSize": 12,
              },
              Object {
                "flex": 1,
                "textAlign": "right",
              },
            ]
          }
        >
          48y
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "rgb(63,50,77)",
                "fontFamily": "Roboto-Regular",
                "fontSize": 15,
              },
              Object {
                "flex": 1,
              },
            ]
          }
        >
          title
        </Text>
      </View>
    </View>
  </View>
  <Image
    resizeMode="cover"
    source={
      Object {
        "testUri": "../../../images/defaultCover2.png",
      }
    }
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 36,
        "justifyContent": "space-between",
        "margin": 20,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../images/iconBotLocation3.png",
          }
        }
        style={
          Object {
            "marginRight": 10,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgb(155,155,155)",
              "fontFamily": "Roboto-Regular",
              "fontSize": 13,
            },
            Object {
              "marginRight": 30,
            },
          ]
        }
      >
        address
         -
         
        98 mi
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "borderColor": "rgba(155,155,155,0.15)",
            "borderRightWidth": 1,
            "flexDirection": "row",
            "paddingHorizontal": 10,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../images/heart.png",
            }
          }
          style={
            Object {
              "marginRight": 10,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "rgb(155,155,155)",
                "fontFamily": "Roboto-Regular",
                "fontSize": 13,
              },
              undefined,
            ]
          }
        >
          0
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingHorizontal": 10,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../images/postGrey.png",
            }
          }
          style={
            Object {
              "marginRight": 10,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "rgb(155,155,155)",
                "fontFamily": "Roboto-Regular",
                "fontSize": 13,
              },
              undefined,
            ]
          }
        >
          0
        </Text>
      </View>
    </View>
  </View>
</View>
`;
